src/main/scala/com/mozilla/telemetry/mozdata/MozData.scala

```diff
@@ -0,0 +1,335 @@
+      logger.warn(s"writing non-catalog global table partition: $detectedUri")
+    }
+
+    log(
+      "writePartition",
+      Map(
+        "detectedUri"->Some(detectedUri),
+        "detectedVersion"->detectedVersion,
+        "owner"->owner,
+        "partition"->partitionString,
+        "sqlTableName"->sqlTableName,
+        "tableName"->Some(tableName),
+        "uri"->uri,
+        "version"->version
+      )
+    )
+
+    extraWriteConfig(writeConfig(df.write)).save(detectedUri)
+
+    // update metadata on global tables
+    if (sqlTableName.isDefined) {
+      metadataUpdateMethods
+        .getOrElse(defaultMetadataUpdateMethods)
+        .foreach(updateMetadata(sqlTableName.get, _, spark))
+    }
+  }
+
+  /** Write table to long term storage
+    *
+    * @param df DataFrame to write
+    * @param tableName table to write
+    * @param version specific version of table, required for global tables,
+    *                defaults to latest or "v1" if latest can't be determined
+    * @param owner optional email that identifies non-global namespace
+    * @param uri optional non-standard location for this table
+    * @param metadataUpdateMethods optional methods to use to update metadata
+    *                               after writing partitioned global tables,
+    *                               WARNING default "sql" method *requires*
+    *                               table to be partitioned
+    * @param extraWriteConfig optional function to configure the DataFrameWriter
+    */
+  def writeTable(df: DataFrame, tableName: String,
+                 version: Option[String] = None, owner: Option[String] = None,
+                 uri: Option[String] = None,
+                 extraWriteConfig: Function[DataFrameWriter[Row],DataFrameWriter[Row]] = {w=>w},
+                 metadataUpdateMethods: Option[List[String]] = None): Unit = {
+    require(
+      version.isDefined || owner.isDefined || uri.isDefined,
+      "version required to write global table"
+    )
+
+    val (detectedUri, detectedVersion, sqlTableName) = getTableInfo(
+      tableName=tableName,
+      version=version,
+      owner=owner,
+      uri=uri,
+      spark=spark,
+      adHocTablesDir=adHocTablesDir,
+      globalTablesDir=globalTablesDir
+    )
+
+    require(
+      detectedUri.isDefined,
+      s"table is not external: ${sqlTableName.getOrElse(tableName)}"
+    )
+
+    if (sqlTableName.isEmpty && owner.isEmpty && uri.isEmpty) {
+      logger.warn(s"writing non-catalog global table: ${detectedUri.get}")
+    }
+
+    log(
+      "writeTable",
+      Map(
+        "detectedUri"->detectedUri,
+        "detectedVersion"->detectedVersion,
+        "owner"->owner,
+        "sqlTableName"->sqlTableName,
+        "tableName"->Some(tableName),
+        "uri"->uri,
+        "version"->version
+      )
+    )
+
+    extraWriteConfig(writeConfig(df.write)).save(detectedUri.get)
+
+    // update metadata on global tables
+    if (sqlTableName.isDefined) {
```

Comment:
It seems to me that `writeTable` should always add the table to the metastore. What is the scenario when this wouldn't be defined?